lib/ansible/modules/system/zfs_permissions.py

```diff
@@ -0,0 +1,271 @@
+    required: true
+  state:
+    description:
+      - Whether to allow (C(present)), or unallow (C(absent)) a permission.
+    required: true
+    choices: [present, absent]
+  users:
+    description:
+      - Users to whom permission(s) should be granted, separated by commas.
+    required: false
+  groups:
+    description:
+      - Groups to whom permission(s) should be granted, separated by commas.
+    required: false
+  everyone:
+    description:
+      - Apply permissions to everyone.
+    required: false
+    default: false
+    choices: ['on','off']
+  permissions:
+    description:
+      - The permission(s) to delegate, separated by commas (required if C(state) is C(present))
+    required: false
+    choices: ['allow','clone','create','destroy',...]
+  local:
+    description:
+      - Apply permissions to C(name) "locally" (C(zfs allow -l))
+    required: false
+    default: null
+    choices: ['on','off']
+  descendents:
+    description:
+      - Apply permissions to C(name)'s descendents (C(zfs allow -d))
+    required: false
+    default: null
+    choices: ['on','off']
+  recursive:
+    description:
+      - Unallow permissions recursively (ignored when C(state) is C(present))
+    required: false
+    default: false
+    choices: ['on','off']
+author: "Nate Coraor (@natefoo)"
+'''
+
+EXAMPLES = '''
+# Grant `zfs allow` and `unallow` permission to the `adm` user with local+descendents scope
+- zfs_permissions: name=rpool/myfs users=adm permissions=allow,unallow
+
+# Grant `zfs send` to everyone, plus the group `backup`
+- zfs_permissions: name=rpool/myvol groups=backup everyone=yes permissions=send
+
+# Grant `zfs send,receive` to users `foo` and `bar` with local scope only
+- zfs_permissions: name=rpool/myfs users=foo,bar permissions=send,receive local=yes
+
+# Revoke all permissions from everyone (permissions specifically assigned to users and groups remain)
+- zfs_permissions: name=rpool/myfs state=absent everyone=yes
+'''
+
+# This module does not return anything other than the standard
+# changed/state/msg/stdout
+RETURN = '''
+
+'''
+
+
+import sys
+
+
+class ZfsPermissions(object):
+    def __init__(self, module):
+        self.module = module
+        self.name = module.params.get('name')
+        self.state = module.params.get('state')
+        self.users = module.params.get('users') or []
+        self.groups = module.params.get('groups') or []
+        self.everyone = module.boolean(module.params.get('everyone'))
+        self.perms = module.params.get('permissions') or []
+        self.recursive = module.boolean(module.params.get('recursive'))
+        self.scope = None
+        self.changed = False
+        self.__current_perms = None
+
+        if self.state == 'present' and not self.perms:
+            self.module.fail_json(msg='The `permissions` option is required for state=present')
+
+        if self.state == 'present' and not (self.users or self.groups or self.everyone):
```

Comment:
use `required_if`

Reply:
This one wasn't possible because there's not a `required_if_any_of`.